src/bootstrap/config.rs

```diff
@@ -637,18 +637,25 @@ impl Config {
         let default = true;
         config.rust_optimize = optimize.unwrap_or(default);
 
-        let default = match &config.channel[..] {
-            "stable" | "beta" | "nightly" => true,
-            _ => false,
-        };
-        config.rust_debuginfo_lines = debuginfo_lines.unwrap_or(default);
-        config.rust_debuginfo_only_std = debuginfo_only_std.unwrap_or(default);
-        config.rust_debuginfo_tools = debuginfo_tools.unwrap_or(false);
-
         let default = debug == Some(true);
-        config.rust_debuginfo = debuginfo.unwrap_or(default);
         config.rust_debug_assertions = debug_assertions.unwrap_or(default);
 
+        // Lines debuginfo is enabled by default for std on named channels.
+        // FIXME: Perhaps CI should do that instead?
```

Comment:
cc @alexcrichton -- I think it makes sense to do as this suggests

Reply:
Yeah this looks good to me, although tweaking CI by updating `src/ci/run.sh` would also be fine